src/josepy/util.py

```diff
@@ -99,12 +94,17 @@ def __eq__(self, other):
         else:
             return NotImplemented
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return '<{0}({1!r})>'.format(self.__class__.__name__, self._wrapped)
 
-    def public_key(self):
+    def public_key(self) -> 'ComparableKey':
         """Get wrapped public key."""
-        return self.__class__(self._wrapped.public_key())
+        if isinstance(self._wrapped, (rsa.RSAPublicKeyWithSerialization,
+                                      ec.EllipticCurvePublicKeyWithSerialization)):
+            return self
+
+        key = self._wrapped.private_numbers().public_numbers.public_key(default_backend())
+        return self.__class__(key)
```

Comment:
I think we can just do this right?
```suggestion
        return self.__class__(self._wrapped.public_key())
```